script.js

```diff
@@ -1 +1,382 @@
+      // eslint-disable-next-line
+      square.addEventListener('click', (event) => {
+        // we will want to pass in the card element so
+        // that we can change how it looks on screen, i.e.,
+        // "turn the card over"
+        squareClick(event.currentTarget, i, j);
+      });
+
+      rowElement.appendChild(square);
+    }
+    boardElement.appendChild(rowElement);
+  }
+
+  return boardElement;
+};
+
+const gameReset = () => {
+  if (canClick) {
+    clearInterval(ref);
+    timer.remove();
+    inputField.value = '';
+    boardElement.innerHTML = '';
+    numPairs = 0;
+    board.splice(0, board.length);
+    reset.remove();
+    winsList.remove();
+    // inputContainer.remove();
+    initGame();
+    inputContainer.remove();
+    gameInfo.classList.remove('outcome');
+    gameInfo.style.backgroundColor = 'white';
+    countdown();
+    gameInfo.innerHTML = `Hello ${playerName}!<br>You have 3 minutes to find all matching pairs of cards in the board below!<br>Click on any of the cards below to begin.`;
+    const boardEl = buildBoardElements(board);
+    timerCon.appendChild(timer);
+    resetContainer.appendChild(reset);
+    boardContainer.appendChild(boardEl);
+    winsList.innerHTML = `Wins: ${playerWins}`;
+    document.body.appendChild(winsList);
+    reset.addEventListener('click', gameReset); }
+};
+
+const submitClick = () => {
+  input = inputField.value;
+  playerName = input;
+  console.log(input);
+  inputContainer.remove();
+  countdown();
+  gameInfo.innerHTML = `Hello ${playerName}!<br>You have 3 minutes to find all matching pairs of cards in the board below!<br>Click on any of the cards below to begin.`;
+  const boardEl = buildBoardElements(board);
+  timerCon.appendChild(timer);
+  resetContainer.appendChild(reset);
+  boardContainer.appendChild(boardEl);
+  document.body.appendChild(winsList);
+  reset.addEventListener('click', gameReset);
+};
+
+// =============== GAME INITIALIZATION ===================
+
+const initGame = () => {
+  gameInfo.innerHTML = 'Enter player name to begin game';
+  gameInfo.classList.add('info');
+  document.body.appendChild(gameInfo);
+
+  gameInfo.appendChild(inputContainer);
+  inputContainer.appendChild(inputField);
+  inputContainer.appendChild(submitButton);
+  submitButton.addEventListener('click', submitClick);
+
+  document.body.appendChild(boardContainer);
+  document.body.appendChild(timerCon);
+  document.body.appendChild(resetContainer);
+  // document.body.appendChild(leaderDiv);
+  // leaderDiv.appendChild(playerList);
+
+  // create this special deck by getting the doubled cards and
+  // making a smaller array that is ( boardSize squared ) number of cards
+  const doubleDeck = makeDeck();
+  const deckSubset = doubleDeck.slice(0, boardSize * boardSize);
+  deck = shuffleCards(deckSubset);
+
+  // deal the cards out to the board data structure
+  for (let i = 0; i < boardSize; i += 1) {
+    console.log(board);
+    board.push([]);
+    console.log(board);
```

Comment:
Please do not leave console.logs in your PRs as we do not want unnecessary code in the master branch where everyone merges their code into. Console logs will make the files longer and more difficult to read.